PracticeGit/conversion.c

```diff
@@ -11,26 +11,52 @@
 // ----------------------------------------------------------------------------------------------------------------------------------------------------------------
 #include <stdio.h>
 
-// Function to convert decimal to Binary
+#define BITS 32
+
+/// <summary>Function to convert decimal to Binary.</summary>
 void DecToBinary (int n) {
-   for (int i = 7; i >= 0; i--) printf ("%d", (n >> i) & 1);
+   for (int i = 15; i >= 0; i--) printf ("%d", (n >> i) & 1);
    printf ("\n");
-}  
+}
 
-// Function to convert decimal to Hexadecimal
+/// <summary>Function to convert decimal to Hexadecimal.</summary>
 void DecToHexadecimal (int n) {
-   unsigned int num = (unsigned int)n & 0xFF;   // To handle negative numbers in two's complement
-   char hexNum[2];  // Array to store hexadecimal number
-   int i = 0;
-   while (num != 0) {   // convert to hexadecimal using modulus division operator
-      int temp = num % 16;
-      if (temp < 10) hexNum[i] = temp + '0';   // Convert to ASCII
-      else hexNum[i] = temp + 'A' - 10;   // Convert to ASCII (A-F)
-      num /= 16;
-      i++;
+   char hexDigits[8];  // Array to store hexadecimal digits
+   int hexIndex = 0;
+   if (n < 0) {   // Handle negative integers by converting to two's complement
+      int temp = -n;     // Convert positive part to binary
+      int binary[BITS];
+      for (int i = 0; i < BITS; i++) binary[i] = (temp >> i) & 1;  // Convert to binary

```

Comment:
Try to implement binary conversion as separate method irrespective of number of bits and try using it here